packages/payments/src/channel-manager.ts

```diff
@@ -388,6 +399,47 @@ export class ChannelManager implements ChannelManagementAPI {
     };
   }
 
+  /**
+   *
+   * @param objectiveIds objectiveIds to sync on
+   * @param message initial message to send to indexer
+   *
+   * 1. exchanges messages until outbox is empty
+   * 2. if any objectives have not finished, waits for a period of time,
+   *    and then sync the in progress objectives
+   */
+  private async ensureObjectives(
+    objectives: DBObjective[],
+    initialMessage: Outgoing
+  ): Promise<ChannelResult[]> {
+    const remaining = new Map(objectives.map((o) => [o.objectiveId, o]));
+
+    objectives.map(({objectiveId}) =>
+      this.wallet.on('objectiveSucceeded', (o) => {
```

Comment:
This creates `n` wallet listeners. You probably only need one listener that checks if the 'objectiveSuceeded` event is for an objective in the `remaining` list.

Reply:
https://github.com/statechannels/graph-payments/pull/30/commits/240f75a9d3c5175f675a007649e0eeec301293b9